packages/core/src/stateUtils.ts

```diff
@@ -1563,15 +1567,25 @@ function resolveAndExecuteActionsWithContext(
           params: actionParams
         }
       });
-      resolvedAction(actionArgs, actionParams);
+      try {
+        resolvedAction(actionArgs, actionParams);
+      } catch (err) {
+        executingCustomAction = false;
+        // rethrow err
+        throw err;
+      }
     }
 
     if (!('resolve' in resolvedAction)) {
       if (actorScope.self._processingStatus === ProcessingStatus.Running) {
+        executingCustomAction = resolvedAction;
         executeAction();
+        executingCustomAction = false;
```

Comment:
why do we reset this both here and in `executeAction`? couldn't everything related to this change be handled by `executeAction`?

Reply:
[e1ec0f2](https://github.com/statelyai/xstate/pull/4876/commits/e1ec0f267cf8a65c333e1ae4370ca08a7c33c779)